code/modules/admin/verbs/freeze.dm

```diff
@@ -38,7 +38,7 @@ GLOBAL_LIST_EMPTY(frozen_atom_list) // A list of admin-frozen atoms.
 		GLOB.frozen_atom_list += src
 
 		var/obj/effect/overlay/adminoverlay/AO = new
-		if(skip_overlays)
+		if(!skip_overlays)
 			overlays += AO
```

Comment:
Instead of doing `if(frozen) return`s in multiple places, you should take advantage of `priority_overlays`. If you do `add_overlay(AO, priority = TRUE)`, it should be added to their priority overlays list which does not get touched unless something calls `cut_overlays(priority = TRUE)`, and nothing currently does.